pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -523,19 +524,18 @@ func (vs *VSphere) NodeAddresses(nodeName k8stypes.NodeName) ([]v1.NodeAddress,
 
 	// retrieve VM's ip(s)
 	for _, v := range mvm.Guest.Net {
-		var addressType v1.NodeAddressType
 		if vs.cfg.Network.PublicNetwork == v.Network {
-			addressType = v1.NodeExternalIP
-		} else {
-			addressType = v1.NodeInternalIP
-		}
-		for _, ip := range v.IpAddress {
-			v1helper.AddToNodeAddresses(&addrs,
-				v1.NodeAddress{
-					Type:    addressType,
-					Address: ip,
-				},
-			)
+			for _, ip := range v.IpAddress {
```

Comment:
There is PR in review from @BaluDontu  touching the same code - https://github.com/kubernetes/kubernetes/pull/45181
Can we merge both PRs to reduce cherry pick the resolve conflict effort?